### PR TITLE
Sitemaps extensions and fixes

### PR DESCRIPTION
**Pull request by kuhnchris**
##### Description
fixes issues with sitemap or missing data

##### Checklist
- [x] code works

##### Affected core subsystem(s)
sitemap, ui

##### Refers/Fixes
Fixes: #3090 
Fixes: #3108 

###### Funder
- [ ] Validated requested changes were made to specification
- [ ] Bounty payout released to the contributor

### CODE REVIEW

File: app/app/sitemaps.py [hunk 1]
```diff
@@ -22,6 +24,20 @@ def location(self, item):
...
+    priority = 0.9
+
+    def items(self):
+        return Grant.objects.all()
+
+    def lastmod(self, obj):
+        return obj.modified_on
+
+    def location(self, item):
+        return "/grants/"+str(item.id)+"/"+item.slug
```
[the remark this answers is not in the document]
Comment (mbeacom) [reply app/app/sitemaps.py]: That's literal string interpolation, yes. XD

File: app/app/sitemaps.py [hunk 2]
```diff
@@ -22,6 +24,20 @@ def location(self, item):
...
+    priority = 0.9
+
+    def items(self):
+        return Grant.objects.all()
+
+    def lastmod(self, obj):
+        return obj.modified_on
+
+    def location(self, item):
+        return f"/grants/{str(item.id)}/{item.slug}"
```
Comment (mbeacom) [inline app/app/sitemaps.py]: You can eliminate the `str()` here.  The f-string formats the value to `str` automatically by default.

File: app/app/sitemaps.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (kuhnchris) [reply app/app/sitemaps.py]: I'm no django/python dev, hence I didn't know, thanks for telling 👍

File: app/app/sitemaps.py (hunk 2 above)
Comment (kuhnchris) [reply app/app/sitemaps.py]: Ah, sure, didn't really figured out all of em, removing it...

File: app/app/sitemaps.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (mbeacom) [reply app/app/sitemaps.py]: @kuhnchris No worries.  My critiques aren't meant to be pointed or negative.  I hope that you take something out of them and expand your toolbox a bit in the process.

Anyway, thanks for making the adjustments.

**Comment by thelostone-mc** [comment]
@mbeacom LGTM ?

**Comment by captnseagraves** [comment]
Looks to address all comments by @mbeacom !

**Comment by kuhnchris** [comment]
Aye, if something is left, please let me know, I got vacation now so I got some spare time on hand :-)
Also, maybe @owocki wants to combine it with the blog-sitemap task?

**Comment by kuhnchris** [comment]
#3090 
#3543 
@thelostone-mc @captnseagraves here, added grants, labs and wagtail. testing can be done on my test platform, pm me on slack for access.

-Chris

File: .circleci/config.yml [hunk 3]
```diff
@@ -17,57 +15,61 @@ jobs:
...
+      - image: circleci/redis:4-alpine3.8
 
     steps:
       - checkout
 
       # Download and cache dependencies
       - restore_cache:
           keys:
-          - deps37-{{ .Branch }}-{{ checksum "requirements/base.txt" }}
+          - deps37-{{ .Branch }}-{{ checksum "requirements/base.txt" }}-alpine
```
[the remark this answers is not in the document]
Comment (kuhnchris) [reply .circleci/config.yml]: Not sure where this is comming from...?

**Comment by thelostone-mc** [comment]
@kuhnchris this PR seems a lil outdated with 860 commits in this PR. Mind opening a fresh PR with just your changes ?

**Comment by kuhnchris** [comment]
you prefer a new PR or want me just to rebase/squash the changes? the "860 changes" are the re-pull of master

**Comment by thelostone-mc** [comment]
@kuhnchris a new PR would be cleaner i guess

**Pull request closed without merging**